mlforecast/distributed/forecast.py

```diff
@@ -325,8 +340,18 @@ def _fit(
             for name, model in self.models.items():
                 trained_model = clone(model).fit(X, y)
                 self.models_[name] = trained_model.model_
+        elif RAY_INSTALLED and isinstance(data, RayDataset):
+            for name, model in self.models.items():
+                X = RayDMatrix(
+                    prep.select_columns(cols=features + [target_col]),
+                    label=target_col,
+                )
```

Comment:
This can be moved out of the for loop.

Reply:
Thanks for catching that. Fixed in the newest commit.🙌